src/normal_form_game.jl

```diff
@@ -53,6 +53,75 @@ function Base.show(io::IO, player::Player)
     Base.print_array(io, player.payoff_array)
 end
 
+# delete_action
+
+"""
+    delete_action(player, action, player_idx)
+
+Return a new Player instance with the action specified by
+`action` deleted from the action set of the player specified by
+`player_idx`.
+
+# Arguments
+
+- `player::Player` : Player instance.
+- `action::PureAction`: the action to be deleted.
+- `player_idx::Int` : Index of the player to delete action for.
+
+# Returns
+
+- `::Player` : Copy of `Player` with the action deleted as specified.
+
+# Reference
+
+https://stackoverflow.com/questions/53637511/how-to-delete-the-specific-row-of-n-dimensional-array-in-julia
+"""
+
+function delete_action(player::Player{N},
+                        action::PureAction, player_idx::Int) where N
+    if -N <= player_idx <= 0
+      player_idx = player_idx + N
+    end
+    sel = Any[Colon() for i in 1:N]
+    sel[player_idx] = setdiff(axes(player.payoff_array, player_idx), action)
+    payoff_array_new = player.payoff_array[sel...]
+    return Player(payoff_array_new)
+end
+
+
+"""
+    delete_action(player, action, player_idx)
+
+Return a new Player instance with the actions specified by
+`action` deleted from the action set of the player specified by
+`player_idx`.
+
+# Arguments
+
+- `player::Player` : Player instance.
+- `action::AbstractVector{<:PureAction}`: the actions to be deleted
+- `player_idx::Int` : Index of the player to delete actions for.
+
+# Returns
+
+- `::Player` : Copy of `Player` with the actions deleted as specified.
+
+#Reference
+
+https://stackoverflow.com/questions/53637511/how-to-delete-the-specific-row-of-n-dimensional-array-in-julia
+"""
+
+function delete_action(player::Player{N},
+                  action::AbstractVector{<:PureAction}, player_idx::Int) where N
+    if -N <= player_idx <= 0
+      player_idx = player_idx + N
+    end
```

Comment:
Do not allow negative indexing in Julia. (The Python version just followed NumPy's convention.)

Reply:
`players_new  = [delete_action(player, action, player_idx-i+1) for (i, player) in enumerate(g.players)]` in the `delete_action` function for `NormalFormGame` may require `player_idx` negative, so I allowed negative indexing.

Should  I write another code which does not make index nonpositive?

Reply:
`player_idx` should be restricted in {1, ..., N}.